django/db/backends/base/schema.py

```diff
@@ -200,6 +200,13 @@ def effective_default(self, field):
                 default = six.binary_type()
             else:
                 default = six.text_type()
+        elif getattr(field, 'auto_now', False) or getattr(field, 'auto_now_add', False):
+            default = timezone.now()  # default for DateTimeField
+            internal_type = field.get_internal_type()
+            if internal_type == 'DateField':
+                default = default.date()
```

Comment:
Since the ticket you referenced hasn't moved forward for 11 months I rather see a fix for this issue here and a deprecation sometime in the future (maybe 1.9 if somebody does it) than having broken code for the next 2 or 3 releases.